client/landing/gutenboarding/components/domain-picker/suggestion-item.tsx

```diff
@@ -23,6 +23,11 @@ const DomainPickerSuggestionItem: FunctionComponent< Props > = ( {
 } ) => {
 	const { __ } = useI18n();
 
+	const domain = suggestion.domain_name;
+	const dotPos = domain.indexOf( '.' );
+	const domainName = domain.slice( 0, dotPos );
+	const domainTld = domain.slice( dotPos );
```

Comment:
it would be **very** helpful here to give a comment indicating an example of the kind of transform you're wanting to apply. these kinds of direct text-manipulation chains can seem a lot more obvious when writing them than when trying to come in later without knowing the shape of possible inputs and trying to guess at what the shape of the expects outputs is supposed to be.

```js
// separate the user subdomain part of the suggestion
// e.g.
//   pleasereviewmypr.wordpress.com -> [ pleasereviewmypr, .wordpress.com ]
//   codingisfun.blog -> [ codingisfun, .blog ]
```

At least with that we have significantly more indication of what the intention of this code is

Reply:
```js
const match = suggestion.match(/([^.]+)\.(.*)/); // custom domain part and remaining provided domain part
if ( ! match ) {
  // we don't know what happened here but we can't render this
  return suggestion?
}
```